hypershift-operator/controllers/hostedcluster/hostedcluster_controller.go

```diff
@@ -2732,3 +2741,228 @@ func reconcileMachineConfigServerService(svc *corev1.Service) error {
+	err := r.Client.Get(ctx, client.ObjectKeyFromObject(controlPlaneNamespace), controlPlaneNamespace)
+	if err != nil {
+		return fmt.Errorf("failed to get control plane namespace: %w", err)
+	}
+
+	// Reconcile machine-approver role
+	role := machineapprover.Role(controlPlaneNamespace.Name)
+	_, err = controllerutil.CreateOrUpdate(ctx, r.Client, role, func() error {
+		return reconcileMachineApproverRole(role)
+	})
+	if err != nil {
+		return fmt.Errorf("failed to reconcile machine-approver role: %w", err)
+	}
+
+	// Reconcile machine-approver service account
+	sa := machineapprover.ServiceAccount(controlPlaneNamespace.Name)
+	_, err = controllerutil.CreateOrUpdate(ctx, r.Client, sa, NoopReconcile)
+	if err != nil {
+		return fmt.Errorf("failed to reconcile machine-approver service account: %w", err)
+	}
+
+	// Reconcile machine-approver role binding
+	rolebinding := machineapprover.RoleBinding(controlPlaneNamespace.Name)
+	_, err = controllerutil.CreateOrUpdate(ctx, r.Client, rolebinding, func() error {
+		return reconcileMachineApproverRoleBinding(rolebinding, role, sa)
+	})
+	if err != nil {
+		return fmt.Errorf("failed to reconcile machine-approver role binding: %w", err)
+	}
+	config := machineapprover.ConfigMap(controlPlaneNamespace.Name)
+	_, err = controllerutil.CreateOrUpdate(ctx, r.Client, config, func() error {
+		return reconcileMachineApproverConfig(config)
+	})
+	if err != nil {
+		return fmt.Errorf("failed to reconcile machine-approver config: %w", err)
+	}
+
+	// The deployment depends on the kubeconfig being reported.
+	if hcp.Status.KubeConfig != nil {
+		// Resolve the kubeconfig secret for machine-approver
+		kubeconfigSecret := &corev1.Secret{
+			ObjectMeta: metav1.ObjectMeta{
+				Namespace: hcp.Namespace,
+				Name:      "service-network-admin-kubeconfig",
+			},
+		}
+		err = r.Client.Get(ctx, client.ObjectKeyFromObject(kubeconfigSecret), kubeconfigSecret)
+		if err != nil {
+			return fmt.Errorf("failed to get hosted controlplane kubeconfig secret %q: %w", kubeconfigSecret.Name, err)
+		}
+
+		// Reconcile machine-approver deployment
+		image := imageMachineApprover
+		if _, ok := hcluster.Annotations[hyperv1.MachineApproverImage]; ok {
+			image = hcluster.Annotations[hyperv1.MachineApproverImage]
+		}
+		deployment := machineapprover.Deployment(controlPlaneNamespace.Name)
+		_, err = controllerutil.CreateOrUpdate(ctx, r.Client, deployment, func() error {
+			return reconcileMachineApproverDeployment(deployment, hcluster, sa, kubeconfigSecret, config, image, imageMachineApproverVersion)
+		})
+		if err != nil {
+			return fmt.Errorf("failed to reconcile machine-approver deployment: %w", err)
+		}
+	}
+
+	return nil
+}
+
+type ClusterMachineApproverConfig struct {
+	NodeClientCert NodeClientCert `json:"nodeClientCert,omitempty"`
+}
+type NodeClientCert struct {
+	Disabled bool `json:"disabled,omitempty"`
+}
+
+func reconcileMachineApproverConfig(cm *corev1.ConfigMap) error {
+	// disable the client cert csr approval
```

Comment:
why?

Reply:
Used kube-controller-manager for client CSR in this PR.

The machine-approver [client csr approval](https://github.com/openshift/cluster-machine-approver#node-client-csr-approval-workflow) requires a lot of hard coded values (from service account related to machine-config-operator?) like the following:
- The user must be the node bootstrapper
   - The username in the CSR must be `system:serviceaccount:openshift-machine-config-operator:node-bootstrapper`
   - The groups in the CSR must be `system:serviceaccounts:openshift-machine-config-operator`, `system:serviceaccounts`, and `system:authenticated`.